python/lsst/ap/association/dia_object.py

```diff
@@ -190,9 +198,27 @@ def append_dia_source(self, input_dia_source_record):
         # them until we are finished adding sources.
         self._updated = False
 
-        self._dia_source_catalog.append(
-            self._dia_source_catalog.getTable().copyRecord(
-                input_dia_source_record))
+        # We need to test that schema of the source to appended lines up with
+        # the schema specified in the catalog of dia_sources associated with
+        # this dia_object. If not we attempt to access the those that overlap
+        # assuming that the schema defined in this dia_object is a subset of
+        # of the input source's schema.
+        input_schema = input_dia_source_record.getSchema()
+        if input_schema == self._dia_source_schema:
+            self._dia_source_catalog.append(
+                self._dia_source_catalog.getTable().copyRecord(
+                    input_dia_source_record))
+        else:
+            tmp_source_record = afwTable.SourceTable.makeRecord(
+                afwTable.SourceTable.make(self._dia_source_schema))
```

Comment:
Why is this `makeRecord(make())`? Can't you just pass the schema to `makeRecord`?

Reply:
You are correct. Since I have a SourceCatalog available I can use the makeRecord method on that to create the empty record object. If I didn't have the catalog sitting around and wanted to create a SourceRecord I would need to first create the SourceTable from the schema which is what this line does.